src/asm_files.cpp

```diff
@@ -265,7 +265,7 @@ std::map<std::string, size_t> parse_map_section(const libbtf::btf_type_data& btf
     std::map<std::string, size_t> map_offsets;
     for (auto& map : parse_btf_map_section(btf_data)) {
         map_offsets.emplace(map.name, map_descriptors.size());
-        map_descriptors.push_back({
+        map_descriptors.emplace_back(EbpfMapDescriptor{
```

Comment:
please let me know if you find some of these changes too invasive for one PR.
I was looking for areas where we risked making extra copies - and tried to find very low-risk changes to let the STL do its work without forcing copies :)

Reply:
I don't see the benefit in emplacing-back a copy instead of copying a temporary.

Reply:
Yeah. I would have preferred to emplace back direct construction of the object. I'll have to disassemble this to see if it's actually copying a temporary or just in-place constructing a single instance.

Reply:
yeah. reverting these.